src/Main Pages/Home.jsx

```diff
@@ -0,0 +1,80 @@
+import { useContext,useEffect,useState } from 'react'
+import { UserContext } from "../UserContext";
+import { Link } from "react-router-dom";
+import { getAllArticles, getTopics,getTopThreeArticlesPerTopic} from '../Api';
+import {mostPopular } from '../utils/utils'
+import { ArticlePage } from './ArticlePage';
+
+export const Home = () => {
+    const {userDetails,setUserDetails} = useContext(UserContext)
+    const [loading, setLoading] = useState(false)
+    const [allArticles, setAllArticles] = useState([])
+    const [topArticle, setTopArticle] = useState({})
+    const [topics, setTopics] = useState([])
+    const [top3perTopic, setTop3PerTopic] = useState({})
+
+    useEffect(()=>{
+        setLoading(true)
+        setAllArticles([])
+        getAllArticles().then(({articles}) => {
+            setLoading(false)
+            setAllArticles(articles)
+            setTopArticle(mostPopular(articles))
+        })
+        
+    },[])
+    useEffect(() => {
+        setTopics([])
```

Comment:
why are we setting topics as an empty array? Is it not initialised as an empty array already at this point?